src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -131,11 +159,33 @@ protected override void PostPersist(ApplicationEngine engine)
 
             // Distribute GAS for committee
 
-            int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)ProtocolSettings.Default.CommitteeMembersCount);
-            var gasPerBlock = GetGasPerBlock(engine.Snapshot);
-            var pubkey = GetCommittee(engine.Snapshot)[index];
-            var account = Contract.CreateSignatureRedeemScript(pubkey).ToScriptHash();
-            GAS.Mint(engine, account, gasPerBlock * CommitteeRewardRatio / 100);
+            if (ShouldRefreshCommittee(engine.Snapshot.PersistingBlock.Index))
+            {
+                int m = ProtocolSettings.Default.CommitteeMembersCount;
+                int n = ProtocolSettings.Default.ValidatorsCount;
+                var gasPerBlock = GetGasPerBlock(engine.Snapshot);
+                var committee = GetCommitteeFromCache(engine.Snapshot);
+                for (var index = 0; index < committee.Count; index++)
+                {
+                    var member = committee.ElementAt(index);
+                    var pubkey = member.PublicKey;
+                    var account = Contract.CreateSignatureRedeemScript(pubkey).ToScriptHash();
+                    GAS.Mint(engine, account, gasPerBlock * CommitteeRewardRatio / 100);
```

Comment:
We have decided to only mint once in each block, right?

Reply:
Yes, reverted